cogito/ghcommitsink.go

```diff
@@ -1,26 +1,27 @@
 package cogito
 
 import (
-	"math/rand"
+	"log/slog"
 	"time"
 
 	"github.com/hashicorp/go-hclog"
 
 	"github.com/Pix4D/cogito/github"
+	"github.com/Pix4D/cogito/internal/hclogslog"
```

Comment:
why are we mixing `log/slog` and `github.com/Pix4D/cogito/internal/hclogslog`?

Reply:
Cogito uses hashicorp/go-hclog to have structured logging.
Since go 1.21, the stdlib has structured logging also, in the log/slog package.
Thus, there is no reason anymore to use hashicorp/go-hclog.

In particular, package cogito/retry is independent from the rest of cogito and is meant to be imported by other tools.

So, if we were to keep using go-hclog also for cogito/retry, we would be imposing go-hclog as a dependency on any tool importing cogito/retry.

To avoid rewriting the whole cogito to use log/slog in this PR, we are using an adapter from go-hclog to log/slog, that allows both to transparently cohexist.

Reply:
> To avoid rewriting the whole cogito to use log/slog in this PR

I didn't say I don't understand the reasoning, I just said I don't like it... I would prefer if this is not kept for a longer period of time.

> we would be imposing go-hclog as a dependency on any tool importing cogito/retry.

Maybe that was not such a bad option to keep the focus on the important stuff in this PR, which is the retry mechanism.
Now its to late for that and I don't want or expect any changes regarding this topic, just trying to understand better some decisions you took.

Good things: 
- I understood the reasoning even before your comment
- Personally, I also don't like unnecessary importing of dependencies and this PR will avoid that for the end user of the retry package.
Bad things: 
- I am in dilemma here: would doing this in steps be such a bad idea? first use `go-hclog ` in retry pkg, then switch everything...

Reply:
In general I agree that a PR should stay as focused as possible. Now, from my point of view, in this case the focus of the PR is not impacted, because the only package using log/slog is cogito/retry, which did not exist before this PR.

On the other hand, the argument I can make for the introduction of go-quicktest/qt is less strong. Although still limited to the new package cogito/retry (so not adding confusing, unrelated diffs to the PR), go-quicktest/qt is also an additional dependency for cogito (until we replace it everywhere).